### PR TITLE
feat(notification): add show progress option and update docs

### LINKED ISSUE #48125: Show progress bar for auto-closing notification
### What problem does this feature solve?

We have configurations available for setting up the duration for notification(s). But if we do so, it doesn't show any progress when it is going to get closed.

I think, adding a feature to render progress using ant-design's own progress will be great.

### What does the proposed API look like?

We can keep the API unmodified, and make it default behavior. Or we can simply add a new field parallel to `duration`, like this:
```
...
duration?: number | null;
showProgress: boolean = true // we pass in the default value as well
...
```
And based on `showProgress` we render the progress bar for notification(s).

### PR DESCRIPTION
**Pull request by yociduo**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #48125 
https://github.com/react-component/notification/pull/322

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Notification support `showProgress` prop |
| 🇨🇳 Chinese | Notification 支持 `showProgress` 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/notification/index.zh-CN.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ demo:
 <code src="./demo/custom-style.tsx">自定义样式</code>
 <code src="./demo/update.tsx">更新消息内容</code>
 <code src="./demo/stack.tsx" version="5.10.0">堆叠</code>
+<code src="./demo/show-progress.tsx">显示进度条</code>
```
Comment (kiner-tang) [inline components/notification/index.zh-CN.md:34]: 这个名字有点歧义，要优化一下

Comment (yociduo) [reply components/notification/index.zh-CN.md:34]: 是有点奇怪，有什么好的建议吗？

**Comment by yociduo** [comment]
> the preview here https://preview-48353-ant-design.surge.sh/components/notification#notification-demo-show-progress doesn't show any progress indicator

Yes, it's work in progress. I'm waiting for the dependence merged. The demo for rc-notification https://notification-git-fork-yociduo-feat-show-progess-react-component.vercel.app/demo/hooks

**Comment by MadCcc** [comment]
image

可以参考这个设计

**Comment by yociduo** [comment]
> image
> 
> 可以参考这个设计

image
更新了，不过需要等rc-notification库更新

File: components/notification/index.en-US.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ To display a notification message at any of the four corners of the viewport. Ty
 <code src="./demo/custom-style.tsx">Customized style</code>
 <code src="./demo/update.tsx">Update Message Content</code>
 <code src="./demo/stack.tsx" version="5.10.0">Stack</code>
+<code src="./demo/show-with-progress.tsx">Show with progress</code>
```
Comment (kiner-tang) [inline components/notification/index.en-US.md:33]:
```suggestion
<code src="./demo/show-with-progress.tsx" version="5.18.0">Show with progress</code>
```

File: components/notification/index.en-US.md [hunk 3]
```diff
@@ -53,6 +54,7 @@ The properties of config are as follows:
 | closeIcon | Custom close icon | ReactNode | true | 5.7.0: close button will be hidden when setting to null or false |
 | description | The content of notification box (required) | ReactNode | - | - |
 | duration | Time in seconds before Notification is closed. When set to 0 or null, it will never be closed automatically | number | 4.5 | - |
+| showProgress | Show progress bar for auto-closing notification. | boolean |  |  |
```
Comment (kiner-tang) [inline components/notification/index.en-US.md:57]:
```suggestion
| showProgress | Show progress bar for auto-closing notification. | boolean |  | 5.18.0 |
```

File: components/notification/index.zh-CN.md [hunk 4]
```diff
@@ -31,6 +31,7 @@ demo:
 <code src="./demo/custom-style.tsx">自定义样式</code>
 <code src="./demo/update.tsx">更新消息内容</code>
 <code src="./demo/stack.tsx" version="5.10.0">堆叠</code>
+<code src="./demo/show-with-progress.tsx">显示进度条</code>
```
Comment (kiner-tang) [inline components/notification/index.zh-CN.md:34]:
```suggestion
<code src="./demo/show-with-progress.tsx" version="5.18.0">显示进度条</code>
```

File: components/notification/index.zh-CN.md [hunk 5]
```diff
@@ -105,6 +106,7 @@ notification.config({
 | bottom | 消息从底部弹出时，距离底部的位置，单位像素 | number | 24 |  |
 | closeIcon | 自定义关闭图标 | ReactNode | true | 5.7.0：设置为 null 或 false 时隐藏关闭按钮 |
 | duration | 默认自动关闭延时，单位秒 | number | 4.5 |  |
+| showProgress | 显示自动关闭通知框的进度条 | boolean |  |  |
```
Comment (kiner-tang) [inline components/notification/index.zh-CN.md:109]:
```suggestion
| showProgress | 显示自动关闭通知框的进度条 | boolean |  | 5.18.0 |
```

**Comment by kiner-tang** [comment]
image
@yociduo lint 报错了，需要处理一下

**Comment by yociduo** [comment]
> image @yociduo lint 报错了，需要处理一下

修复了，不过需要先处理组件那边的更新。

**Comment by kiner-tang** [comment]
> > image @yociduo lint 报错了，需要处理一下
> 
> 修复了，不过需要先处理组件那边的更新。

我去催一下 rc-notification 那边

File: components/notification/style/index.ts [hunk 6]
```diff
@@ -154,6 +158,39 @@ export const genNoticeStyle = (token: NotificationToken): CSSObject => {
...
+      '&, &::-webkit-progress-bar': {
+        borderRadius: borderRadiusLG,
+        backgroundColor: `rgba(0, 0, 0, 0.04)`,
+      },
+
+      [`&::-moz-progress-bar`]: {
+        backgroundColor: colorPrimary,
+      },
+
+      [`&::-webkit-progress-value`]: {
```
Comment (afc163) [inline components/notification/style/index.ts:188]: 不加 webkit 和 moz 可行么。

Comment (yociduo) [reply components/notification/style/index.ts:188]: 不可以，chrome和ff都不工作

File: components/notification/style/index.ts [hunk 7]
```diff
@@ -34,6 +34,7 @@ export interface NotificationToken extends FullToken<'Notification'> {
   notificationMarginBottom: number;
   notificationMarginEdge: number;
   notificationStackLayer: number;
+  notificationProgressBarSize: number;
```
Comment (afc163) [inline components/notification/style/index.ts:37]:
```suggestion
  notificationProgressHeight: number;
```

File: components/notification/style/index.ts [hunk 8]
```diff
@@ -154,6 +158,39 @@ export const genNoticeStyle = (token: NotificationToken): CSSObject => {
...
+      inlineSize: `calc(100% - ${unit(borderRadiusLG)} * 2)`,
+      left: {
+        _skip_check_: true,
+        value: borderRadiusLG,
+      },
+      right: {
+        _skip_check_: true,
+        value: borderRadiusLG,
+      },
+      bottom: calc(notificationProgressBarSize).mul(-1).equal(),
```
Comment (afc163) [inline components/notification/style/index.ts:175]:
```suggestion
      bottom: 0,
```

放卡片里面会不会比较好。

File: components/notification/style/index.ts [hunk 9]
```diff
@@ -154,6 +157,39 @@ export const genNoticeStyle = (token: NotificationToken): CSSObject => {
...
+      blockSize: notificationProgressHeight,
+      border: 0,
+
+      '&, &::progress-bar': {
+        borderRadius: borderRadiusLG,
+        backgroundColor: `rgba(0, 0, 0, 0.04)`,
+      },
+
+      [`&::-moz-progress-bar`]: {
+        backgroundColor: colorPrimary,
```
Comment (afc163) [inline components/notification/style/index.ts:184]: 设计稿里是个渐变色。

File: components/notification/style/index.ts [hunk 10]
```diff
@@ -154,6 +156,39 @@ export const genNoticeStyle = (token: NotificationToken): CSSObject => {
...
+        backgroundColor: `rgba(0, 0, 0, 0.04)`,
+      },
+
+      [`&::-moz-progress-bar`]: {
+        background: `linear-gradient(90deg, #31afff, #1677ff)`,
+      },
+
+      [`&::-webkit-progress-value`]: {
+        borderRadius: borderRadiusLG,
+        background: `linear-gradient(90deg, #31afff, #1677ff)`,
```
Comment (afc163) [inline components/notification/style/index.ts:188]: #1677ff 可以用主色

Comment (afc163) [reply components/notification/style/index.ts:188]: 找找 #31afff 和主色的关系？或者在 blue 里找一个类似的。

File: components/notification/index.en-US.md [hunk 11]
```diff
@@ -53,6 +54,7 @@ The properties of config are as follows:
 | closeIcon | Custom close icon | ReactNode | true | 5.7.0: close button will be hidden when setting to null or false |
 | description | The content of notification box (required) | ReactNode | - | - |
 | duration | Time in seconds before Notification is closed. When set to 0 or null, it will never be closed automatically | number | 4.5 | - |
+| showProgress | Show progress bar for auto-closing notification. | boolean |  | 5.18.0 |
```
Comment (kiner-tang) [inline components/notification/index.en-US.md:57]:
```suggestion
| showProgress | Show progress bar for auto-closing notification | boolean |  | 5.18.0 |
```

**Review by kiner-tang — APPROVED** [review]

**Comment by kiner-tang** [comment]
@SocketSecurity ignore npm/@ant-design/tools@18.0.2
@SocketSecurity ignore npm/core-js@3.34.0
@SocketSecurity ignore npm/@biomejs/biome@1.7.3

**Pull request merged**